spaceship.zsh

```diff
@@ -370,9 +370,9 @@ spaceship_golang_version() {
 
   command -v go > /dev/null 2>&1 || return
 
-  local gofiles=(*.go(N))
+  setopt EXTENDED_GLOB LOCAL_OPTIONS
```

Comment:
I've added the comment and also fixed a crucial typo 😓 I made on the previous commit.